tests/test_grid.py

```diff
@@ -99,6 +99,47 @@ def test_concatenate_coordinates(self):
         fd = fd1.concatenate(fd2, as_coordinates=True)
         np.testing.assert_equal(None, fd.axes_labels)
 
+    def test_concatenate_samples(self):
+        fd1 = FDataGrid([[1, 2, 3, 4, 5], [2, 3, 4, 5, 6]])
```

Comment:
Add a test that passes a generator expression and make that work.

Reply:
Changed